### PR TITLE
Day3 part2

### PR DESCRIPTION
**Pull request by PascalWo**
Task:

In the first part, you should calculate the inches which are claimed by two or more rectangles.
Now you should find the id of the single rectangle which has no conflict with others rectangles.

I refactroed a function from part 1 to not produce duplictaed code
Also I used for-loops because I had problems to build streams
This was mainly because of the second condition in the loop (boolean stillFree = true)

Maybe their is a better way?

### CODE REVIEW

File: src/main/java/org/haffson/adventofcode/days/day03/Day03.java [hunk 1]
```diff
@@ -86,4 +80,37 @@ List<RectangleClaim> convertStringToRectangleList(final List<String> stringList)
...
+                    }else if (x == claim.getRightXCoordinate() - 1 && y == claim.getBottomYCoordinate() - 1){
+                        intactClaimId[0] = Integer.parseInt(claim.getId());
+                    }
+                }
+            }
+        });
+        return intactClaimId[0];
+    }
+
+    private String[][] get2DClaimArray(final List<RectangleClaim> claimList){
```
Comment (mfbieber) [inline src/main/java/org/haffson/adventofcode/days/day03/Day03.java:105]: What about adding more tests to test the edge cases? I mean also the not "happy case". Think about how to secure your implementation, e.g. when someone joins your project and starts to refactor your method. What can you think of?

File: src/test/java/org/haffson/adventofcode/days/day03/Day03Test.java [hunk 2]
```diff
@@ -72,4 +72,25 @@ public void testConvertStringToRectangleList_returnListOfRectangles() {
         //assert
         Assert.assertEquals(expectedResult, actualResult);
     }
+
+    @Test
+    public void test_secondPart_returnsExpectedResult() {
+        //arrange
+        final Day03 day03 = new Day03(fileReaders);
+        when(fileReaders.getInputList("src/main/resources/puzzle_input/day3_input.txt"))
+                .thenReturn(new ArrayList<>(Arrays.asList(
```
[the remark this answers is not in the document]
Comment (PascalWo) [reply src/test/java/org/haffson/adventofcode/days/day03/Day03Test.java:81]: not possible with java 8 :( I will use it after the java 17 update :)
